File: example/src/main/java/com/matnsolutions/matlive/lib/widgets/seat_action_bottom_sheet.dart
import 'package:flutter/material.dart';
import 'package:matlive_client_sdk_flutter/audio/define/mat_live_user.dart';

class SeatActionBottomSheet extends StatelessWidget {
  final MatLiveUser? user;
  final Function()? onTakeMic;
  final Function()? onMuteMic;
  final Function()? onRemoveSpeaker;
  final Function()? onLeaveMic;
  final Function()? onUnLockMic;
  final Function()? onLockMic;
  final Function()? onSwitch;

  const SeatActionBottomSheet({
    super.key,
    required this.user,
    this.onTakeMic,
    this.onMuteMic,
    this.onRemoveSpeaker,
    this.onLeaveMic,
    this.onUnLockMic,
    this.onLockMic,
    this.onSwitch,
  });

  @override
  Widget build(BuildContext context) {
    return Container(
      padding: const EdgeInsets.symmetric(vertical: 16),
      decoration: const BoxDecoration(
        color: Colors.white,
        borderRadius: BorderRadius.vertical(top: Radius.circular(16)),
      ),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Container(
            width: 40,
            height: 4,
            margin: const EdgeInsets.only(bottom: 16),
            decoration: BoxDecoration(
              color: Colors.grey[300],
              borderRadius: BorderRadius.circular(2),
            ),
          ),
          if (user == null) ...[
            _ActionButton(
              icon: Icons.mic,
              label: 'Take Mic',
              onTap: () {
                Navigator.pop(context);
                onTakeMic?.call();
              },
            ),_ActionButton(
              icon: Icons.swap_calls,
              label: 'Switch Mic',
              onTap: () {
                Navigator.pop(context);
                onSwitch?.call();
              },
              isDestructive: true,
            ),
            _ActionButton(
              icon: Icons.lock_outline,
              label: 'Lock Mic',
              onTap: () {
                Navigator.pop(context);
                onLockMic?.call();
              },
              isDestructive: true,
            ),
            _ActionButton(
              icon: Icons.lock_open,
              label: 'UnLock Mic',
              onTap: () {
                Navigator.pop(context);
                onUnLockMic?.call();
              },
              isDestructive: true,
            ),
          ] else ...[
            ValueListenableBuilder<bool>(
                valueListenable: user!.isMicOnNotifier,
                builder: (context, isMuted, _) {
                  return isMuted
                      ? _ActionButton(
                          icon: Icons.mic_off,
                          label: 'Mute Mic',
                          onTap: () {
                            Navigator.pop(context);
                            onMuteMic?.call();
                          },
                        )
                      : _ActionButton(
                          icon: Icons.mic,
                          label: 'UnMute Mic',
                          onTap: () {
                            Navigator.pop(context);
                            onMuteMic?.call();
                          },
                        );
                }),
            _ActionButton(
              icon: Icons.person_remove,
              label: 'Remove Speaker',
              onTap: () {
                Navigator.pop(context);
                onRemoveSpeaker?.call();
              },
              isDestructive: true,
            ),

            _ActionButton(
              icon: Icons.exit_to_app,
              label: 'Leave Mic',
              onTap: () {
                Navigator.pop(context);
                onLeaveMic?.call();
              },
              isDestructive: true,
            ),
          ],
        ],
      ),
    );
  }
}

class _ActionButton extends StatelessWidget {
  final IconData icon;
  final String label;
  final VoidCallback onTap;
  final bool isDestructive;

  const _ActionButton({
    required this.icon,
    required this.label,
    required this.onTap,
    this.isDestructive = false,
  });

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onTap,
      child: Container(
        width: double.infinity,
        padding: const EdgeInsets.symmetric(vertical: 12, horizontal: 16),
        child: Row(
          children: [
            Icon(
              icon,
              color: isDestructive ? Colors.red : Colors.black87,
              size: 24,
            ),
            const SizedBox(width: 12),
            Text(
              label,
              style: TextStyle(
                fontSize: 16,
                color: isDestructive ? Colors.red : Colors.black87,
                fontWeight: FontWeight.w500,
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: example/src/main/java/com/matnsolutions/matlive/lib/widgets/SeatActionBottomSheet.kt
package com.matnsolutions.matlive.lib.widgets

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.matnsolutions.matlive_sdk.audio.define.MatLiveUser

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SeatActionBottomSheet(
    showBottomSheet: Boolean,
    user: MatLiveUser?,
    onTakeMic: (() -> Unit)? = null,
    onMuteMic: (() -> Unit)? = null,
    onRemoveSpeaker: (() -> Unit)? = null,
    onLeaveMic: (() -> Unit)? = null,
    onLockMic: (() -> Unit)? = null,
    onUnLockMic: (() -> Unit)? = null,
    onSwitch: (() -> Unit)? = null,
    onDismiss: () -> Unit,
) {
    if (showBottomSheet) {
        ModalBottomSheet(
            onDismissRequest = onDismiss,
            containerColor = Color.White,
            dragHandle = {
                Box(
                    modifier = Modifier
                        .width(40.dp)
                        .height(4.dp)
                        .background(Color.Gray.copy(alpha = 0.3f), shape = RoundedCornerShape(2.dp))
                )
            }
        ) {
            Column(
                modifier = Modifier.padding(vertical = 16.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                if (user == null) {
                    ActionButton(
                        icon = Icons.Filled.Edit,
                        label = "Take Mic",
                        onTap = {
                            onTakeMic?.invoke()
                            onDismiss()
                        }
                    )
                    ActionButton(
                        icon = Icons.Filled.Info,
                        label = "Switch Mic",
                        onTap = {
                            onSwitch?.invoke()
                            onDismiss()
                        },
                        isDestructive = true
                    )
                    ActionButton(
                        icon = Icons.Filled.Lock,
                        label = "Lock Mic",
                        onTap = {
                            onLockMic?.invoke()
                            onDismiss()
                        },
                        isDestructive = true
                    )
                    ActionButton(
                        icon = Icons.Filled.AccountBox,
                        label = "UnLock Mic",
                        onTap = {
                            onUnLockMic?.invoke()
                            onDismiss()
                        },
                        isDestructive = true
                    )
                } else {
                    val isMicOn by user.isMicOnNotifier.observeAsState(initial = false)
                    if (isMicOn) {
                        ActionButton(
                            icon = Icons.Filled.MicOff,
                            label = "Mute Mic",
                            onTap = {
                                onMuteMic?.invoke()
                                onDismiss()
                            }
                        )
                    } else {
                        ActionButton(
                            icon = Icons.Filled.Mic,
                            label = "UnMute Mic",
                            onTap = {
                                onMuteMic?.invoke()
                                onDismiss()
                            }
                        )
                    }
                    ActionButton(
                        icon = Icons.Filled.Delete,
                        label = "Remove Speaker",
                        onTap = {
                            onRemoveSpeaker?.invoke()
                            onDismiss()
                        },
                        isDestructive = true
                    )
                    ActionButton(
                        icon = Icons.Filled.ExitToApp,
                        label = "Leave Mic",
                        onTap = {
                            onLeaveMic?.invoke()
                            onDismiss()
                        },
                        isDestructive = true
                    )
                }
                Spacer(modifier = Modifier.height(16.dp))
            }
        }
    }
}

@Composable
private fun ActionButton(
    icon: ImageVector,
    label: String,
    onTap: (() -> Unit)?,
    isDestructive: Boolean = false
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(enabled = onTap != null) { onTap?.invoke() }
            .padding(vertical = 12.dp, horizontal = 16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Icon(
            imageVector = icon,
            contentDescription = label,
            tint = if (isDestructive) Color.Red else Color.Black.copy(alpha = 0.87f),
            modifier = Modifier.size(24.dp)
        )
        Spacer(modifier = Modifier.width(12.dp))
        Text(
            text = label,
            style = TextStyle(
                fontSize = 16.sp,
                color = if (isDestructive) Color.Red else Color.Black.copy(alpha = 0.87f),
            )
        )
    }
}